CRM/Core/BAO/Domain.php

```diff
@@ -192,19 +192,18 @@ public static function multipleDomains() {
    * @throws \CRM_Core_Exception
    */
   public static function getNameAndEmail($skipFatal = FALSE, $returnString = FALSE) {
-    $fromEmailAddress = CRM_Core_OptionGroup::values('from_email_address', NULL, NULL, NULL, ' AND is_default = 1');
+    $fromEmailAddress = \Civi\Api4\DomainEmailAddress::get(FALSE)
+      ->addSelect('name', 'email')
+      ->addWhere('domain_id', '=', 'current_domain')
+      ->addWhere('is_default', '=', TRUE)
+      ->addWhere('is_active', '=', TRUE)
+      ->execute()->first();
     if (!empty($fromEmailAddress)) {
       if ($returnString) {
         // Return a string like: "Demonstrators Anonymous" <info@example.org>
-        return $fromEmailAddress;
+        return [CRM_Utils_Mail::formatFromAddress($fromEmailAddress)];
```

Comment:
This returns an array with a single element. Is that intentional? The comment and variable name imply that a string is returned, not an array.

Reply:
The 2nd param to this function was pretty weird & the comments weren't very helpful.
I've pushed a commit to deprecate it.